app/controllers/admin/dashboard_controller.rb

```diff
@@ -9,10 +9,10 @@ def index
 
       @system_checks         = Admin::SystemCheck.perform(current_user)
       @time_period           = (29.days.ago.to_date...Time.now.utc.to_date)
-      @pending_users_count   = User.pending.count
-      @pending_reports_count = Report.unresolved.count
-      @pending_tags_count    = Tag.pending_review.count
-      @pending_appeals_count = Appeal.pending.count
+      @pending_users_count   = User.pending.async_count
+      @pending_reports_count = Report.unresolved.async_count
+      @pending_tags_count    = Tag.pending_review.async_count
+      @pending_appeals_count = Appeal.pending.async_count
```

Comment:
It may even make sense to perform those before the `@system_checks` which might themselves take a bit of time. Doing so would let the queries execute during the system checks.

Reply:
Indeed -- will rebase along those lines whenever a Rails version which fixes this feature is released.